serial_views.py

```diff
@@ -38,45 +51,120 @@ class Serializer():
+        #     print(request.POST)
+        #     # lat = float(request.POST.get('lat'))
+        #     # long = float(request.POST.get('long'))
+        #     region_id = str(request.POST.get('long'))
+        #     print(region_id)
+        #
+        #     if region_id in Serializer.gj['features']['properties']['name']:
+        #         region_boundary = Serializer.gj['features']['geometry']['coordinates']
+        #         geometry = MultiPolygon(region_boundary)
+        #         # geometry = loadswkb(str(record.Series.location.point), True)
+        #         # feature = Feature(id=record.Series.id, geometry=geometry)
+        #         feature = Feature(id=102, geometry=geometry)
+        #         self.features.append(feature)
+        #
+        # elif request.method == 'GET':
+        #     print(request.GET)
+        myfeatures = []
+        region_id = str(request)
+        # stores the current region boundary
+        res_powerplant_tbl = oep_models.ego_dp_res_classes['ResPowerPlant']
+
+        wkbs = []
+        for f in Serializer.gj['features']:
+            if region_id in f['properties']['name']:
+                region_boundary = f['geometry']['coordinates']
+                boundary_geometry = MultiPolygon(region_boundary)
+                # create shapely geometry from geojson feature
+                _geom = shape(boundary_geometry)
+                # convert shaply.geometry to wkbelement
+                # query_geom = from_shape(_geom, srid=4326)
+
+                # wkbs.append(_geom)
+                # wkbs.append(from_shape(_geom, srid=3035))
+                wkbs.append(from_shape(_geom, srid=4326))
+
+        if LOCAL_TESTING is False:
+            # Query the DB with the given wkbelement as input
+            for wkb in wkbs:
+                tbl_cols = Bundle('powerplant', res_powerplant_tbl.id, res_powerplant_tbl.geom,
+                                  res_powerplant_tbl.generation_type, res_powerplant_tbl.scenario)
+
+                # ToDo: Insert dropdown selection in the filter options like 'solar'
+                # ToDo: Change the geom column to rea_geom_new mind there is another srid 3035 in this column
+                for record in Serializer.session.query(tbl_cols)\
+                        .filter(and_(tbl_cols.c.geom.ST_Within(wkb), tbl_cols.c.scenario == 'Status Quo',
+                                     tbl_cols.c.generation_type == 'solar')).limit(1000):
+
+                    region_contains = loadswkb(str(record.powerplant.geom), True)
+
+                    # feature_prop = Feature(id=record.powerplant.id, property='')
+                    feature = Feature(id=record.powerplant.id, geometry=region_contains, property='')
+                    myfeatures.append(feature)
+
+                    # print('WAIT')
+
+        return HttpResponse(dumps(FeatureCollection(myfeatures)), content_type="application/json")
+
+    def ppr_popup_view(self, request):
         """
```

Comment:
This method is not used, no? There is the function ppr_popup_view defined below. I would suggest to delete the unused code to avoid confusion